# **Analyzing Africa’s Conflict Landscape to Inform Buisness Investments**

![Sample Image](RTX1DRV2.jpg)

### **BUSINESS UNDERSTANDING**
<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
Africa is highly marketable to multinationals due to its vast natural resources, growing consumer market, and expanding economic opportunities. However conflict and Political instability can lead to supply chain disruptions, regulatory hurdles, and heightened security expenses, making it difficult for businesses to operate efficiently. Additionally, conflicts often result in workforce displacement, financial volatility, and reputational risks.  Key Risk Factors regarding the conflict such as  event forecasting, conflict risk Score, conflict risk classification are key considerations that multinational organizations are going to be interested in when deciding which country to invest in and also when crafting their risk management strategies. This project will examine these factors using a structured data driven approach. We will use a data set from https://acleddata.com/about-acled/  . Using this data we shall;<br>
        - Investigate insights on key risk factors causing the conflict, event forecasting, conflict risk score, and risk classification.<br>
        - Analyze the data set to identify conflict prone areas, forecast conflict areas, establish conflict score for countries, risk classification for different countries.<br>
        - Perform statistical tests to understand which areas are prone to conflict, their risk scores, forecast areas in future likely to experience conflict and classify different types of conflict.<br>
        - Provide actionable recommendations to multinationals on conflicts in Africa in terms of areas prone to conflicts and types of conflict so they can formulate good risk management strategies.</p>

### **PROBLEM STATEMENT**
<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
Our data science consulting firm has been hired by a UK based multinational organization (VMagic Leather Processing Factory) that is interested in Africa. The mission of the firm is to establish a manufacturing plant in a location that is less prone to conflict and political instability. The company has tasked us to analyze data for different countries in Africa and come up with a model that is going to help them decide which country to invest in. Our goal is to formulate a model that can accurately determine conflict hot spots, come up with a risk score, and forecast conflict events.</p>
 
### **OBJECTIVES**
<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
1. Conflict Analysis in Africa: Establish a foundational understanding of conflict types, high-risk regions, and underlying causes by analyzing relevant data.<br>
2. Dataset Exploration: Gain familiarity with the dataset’s structure, terminology, and quality by examining key variables, including numerical and categorical data.<br>
3. Business Insights for Site Selection: Assist a multinational client in identifying conflict-free locations for a manufacturing facility by:<br>
- Developing a predictive model to forecast conflict occurrences based on historical data and socio-political factors.<br>
- Identifying key conflict drivers through correlation analysis of variables such as fatalities, disorder types, and actor involvement.<br>
- Assessing spatial and temporal trends to understand conflict variations across regions and time periods.<br>
- Evaluating conflict severity by predicting the likelihood of high-fatality events.<br>
- Enhancing early warning systems through geographic risk assessments.<br>
4. Statistical and Predictive Analysis<br>
- Conduct descriptive and exploratory analysis using measures such as mean, median, variance, and frequency distribution.<br>
- Perform correlation analysis to examine relationships between key variables.<br>
- Apply hypothesis testing, including ANOVA, to compare disorder types across locations.<br>
- Use predictive modeling techniques like logistic regression to assess conflict probabilities.<br>
-Conduct trend analysis to visualize conflict patterns over time.</p>

#### **Summary and Recommendations**

<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
Synthesize findings and provide strategic recommendations, including identifying the most suitable country for investment and highlighting conflict-driving factors.</p>

### **DATA UNDERSTANDING**
<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
In this project, we worked with the ACLED dataset—a comprehensive repository of real-time data on political violence, protests, and other politically relevant events from around the world. The dataset captures a wide array of details, including unique event identifiers, precise dates, event types and sub-types, information about the actors involved, geographic coordinates, and additional contextual attributes such as reported fatalities, tags, and sources. The raw data is collected via a global network that aggregates reports from local, regional, and national media, government publications, and over 50 local data collection partners, covering more than 100 languages and updated on a weekly basis. Data can be accessed through ACLED’s export tools, curated downloads, or an API, and is accompanied by comprehensive documentation that clearly defines each feature, ensuring analytical clarity
The dataset is extensive with 413,948 rows and 31 features.<br>
Below is a table summarizing the key features of the ACLED dataset along with their meanings:</p>

| Feature           | Description |
|-------------------|------------|
| event_id_cnty     | A unique alphanumeric identifier combining a numeric ID with a country code (e.g., ETH9766). |
| event_date        | The date on which the event occurred, recorded in YYYY-MM-DD format. |
| year              | The year during which the event took place. |
| time_precision    | A numeric code (1–3) indicating the precision of the event date, with 1 being the most precise. |
| disorder_type     | The overarching category of the event, such as Political violence, Demonstrations, or Strategic developments. |
| event_type        | The general classification of the event (e.g., Battles, Protests, Explosions/Remote violence). |
| sub_event_type    | A more specific classification within the event type (e.g., Armed clash, Peaceful protest). |
| actor1            | The primary actor involved in the event. |
| assoc_actor_1     | Additional or supporting actor(s) associated with actor1. |
| inter1            | A categorical code describing actor1’s type (e.g., Rebel group, State forces). |
| actor2            | The secondary actor involved in the event (which may represent a target or another involved party). |
| assoc_actor_2     | Additional or supporting actor(s) associated with actor2. |
| inter2            | A categorical code describing actor2’s type. |
| interaction       | A combined description derived from inter1 and inter2, indicating the nature of the interaction between the actors. |
| civilian_targeting| An indicator specifying whether civilians were the primary target (e.g., “Civilians targeted”). |
| iso               | A three-digit ISO numeric code representing the country where the event occurred. |
| region            | The geographic region in which the event took place (e.g., Eastern Africa). |
| country           | The country in which the event occurred. |
| admin1            | The primary sub-national administrative region (e.g., state or province). |
| admin2            | The secondary sub-national administrative region (if applicable). |
| admin3            | The tertiary sub-national administrative region (if available). |
| location          | The specific name of the place where the event occurred. |
| latitude          | The latitude coordinate of the event location in decimal degrees. |
| longitude         | The longitude coordinate of the event location in decimal degrees. |
| geo_precision     | A numeric code (1–3) indicating the precision of the geographic data, with 1 being the most precise. |
| source            | The source(s) used to report the event, which may include multiple sources separated by semicolons. |
| source_scale      | Indicates the geographic closeness of the source to the event (e.g., Local partner, National). |
| notes             | Additional descriptive information about the event. |
| fatalities        | The number of reported fatalities resulting from the event (0 if none are reported). |
| tags              | Keywords or structured tags that provide additional context to the event (e.g., “women targeted: politicians”). |
| timestamp         | A Unix timestamp representing when the event was uploaded to the ACLED API, capturing the exact date and time of data entry. |

<p style="text-align: justify; font-size:16px; color:rgb(1, 1, 1);line-height: 1.8;">
Furthermore, ACLED categorizes its events using a detailed classification system outlined in Table 2. This table presents the different event types, their associated sub-event types, and the overarching disorder categories. The event type provides a broad categorization—such as Battles or Protests—while the sub-event types offer more granular details (for example, distinguishing between an “Armed clash” and a “Government regains territory” within Battles). The disorder type further contextualizes the event by indicating whether it falls under political violence, demonstrations, or strategic developments. This structured taxonomy is essential for accurately analyzing conflict dynamics and understanding the nuanced nature of political violence.</p>

| Event Type                 | Sub-Event Types                                                                                                                                   | Disorder Type                    |
|---------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------|
| Battles                   | Government regains territory, Non-state actor overtakes territory, Armed clash                                                                      | Political violence               |
| Protests                  | Excessive force against protesters, Protest with intervention, Peaceful protest                                                                    | Political violence; Demonstrations |
| Riots                     | Violent demonstration, Mob violence                                                                                                                | Political violence; Demonstrations |
| Explosions/Remote violence| Chemical weapon, Air/drone strike, Suicide bomb, Shelling/artillery/missile attack, Remote explosive/landmine/IED, Grenade                         | Political violence               |
| Violence against civilians| Sexual violence, Attack, Abduction/forced disappearance                                                                                            | Political violence               |
| Strategic developments    | Agreement, Arrests, Change to group/activity, Disrupted weapons use, Headquarters or base established, Looting/property destruction, Non-violent transfer of territory, Other | Strategic developments           |






In [18]:
# Importing necessary libraries
import pandas as pd
import numpy as np
import zipfile
import csv

# Data visualization
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

# performance metrics
from scipy import stats
from scipy.stats import norm

from sklearn.preprocessing import StandardScaler
# Filtering future warnings
import warnings
warnings.filterwarnings('ignore')

#### **Loading the Dataset**

In [19]:
# Path to your ZIP file
zip_path = '../Africa_1997-2025_Feb28.zip'

# Open the zip and read CSV inside
with zipfile.ZipFile(zip_path) as z:
    # List files inside the zip
    print(z.namelist())
    
    # Assuming there's a CSV file inside, read it directly
    with z.open('Africa_1997-2025_Feb28.csv') as f:
        df = pd.read_csv(f)

['Africa_1997-2025_Feb28.csv', '__MACOSX/._Africa_1997-2025_Feb28.csv']


#### **Exploring the structure of the Dataset**

In [30]:
print("Dataset Overview:")
# checking the shape of the dataset
print(f"This dataset contains {df.shape[0]} rows and {df.shape[1]} columns\n")

# Checking the Columns
print(f"This are the columns available in the dataset {df.columns} columns")

#Checking the Data Type
print(f"The shows more information about the dataset, that is the entries and data type per column\n")
print(df.info())  # Column names, data types, and non-null counts

# Preview the first few rows of data
print("\nFirst 5 Rows:")
df.head()



Dataset Overview:
This dataset contains 413947 rows and 27 columns

This are the columns available in the dataset Index(['event_date', 'year', 'disorder_type', 'event_type', 'sub_event_type',
       'actor1', 'assoc_actor_1', 'inter1', 'actor2', 'assoc_actor_2',
       'inter2', 'interaction', 'civilian_targeting', 'iso', 'region',
       'country', 'admin1', 'admin2', 'admin3', 'location', 'latitude',
       'longitude', 'geo_precision', 'source', 'source_scale', 'fatalities',
       'timestamp'],
      dtype='object') columns
The shows more information about the dataset, that is the entries and data type per column

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413947 entries, 0 to 413946
Data columns (total 27 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   event_date          413947 non-null  object 
 1   year                413947 non-null  int64  
 2   disorder_type       413947 non-null  object 
 3   event_ty

event_date  year           disorder_type              event_type  \
0  2025-02-26  2025          Demonstrations                Protests   
1  2025-02-25  2025          Demonstrations                Protests   
2  2025-02-24  2025  Strategic developments  Strategic developments   
3  2025-02-20  2025      Political violence                 Battles   
4  2025-02-20  2025          Demonstrations                Protests   

     sub_event_type                              actor1 assoc_actor_1  \
0  Peaceful protest                Protesters (Algeria)           NaN   
1  Peaceful protest                Protesters (Algeria)           NaN   
2           Arrests    Police Forces of Algeria (2019-)           NaN   
3       Armed clash  Military Forces of Algeria (2019-)           NaN   
4  Peaceful protest                Protesters (Algeria)           NaN   

         inter1                                 actor2  \
0    Protesters                                    NaN   
1    Protesters                                    NaN   
2  State forces                    Civilians (Algeria)   
3  State forces  AQIM: Al Qaeda in the Islamic Maghreb   
4    Protesters                                    NaN   

                                       assoc_actor_2  ...        admin2  \
0                                                NaN  ...       El Tarf   
1                                                NaN  ...  Had Echkalla   
2  CNAPESTE: National Independent Council of High...  ...         Msila   
3                                                NaN  ...   Ouled Antar   
4                                                NaN  ...        Djanet   

  admin3          location  latitude longitude geo_precision  \
0    NaN           El Tarf   36.7686    8.3167             1   
1    NaN      Had Echkalla   35.6812    1.1467             1   
2    NaN            M'Sila   35.7058    4.5419             1   
3    NaN  Changoura Forest   36.0458    2.7012             1   
4    NaN        Tin Alkoum   24.6192   10.2125             1   

                                       source source_scale fatalities  \
0                                   Akher Saa     National          0   
1                                    Echorouk     National          0   
2                            Maghreb Emergent     Regional          0   
3  Algeria Press Service; El Khabar (Algeria)     National          1   
4                                    Echorouk     National          0   

    timestamp  
0  1741039795  
1  1741039795  
2  1741039795  
3  1740436907  
4  1740447536  

[5 rows x 27 columns]

In [32]:
# Checking the Value counts for Categorical columns eg. 'region', 'country', 'event_type', 'disorder_type'
print("\nValue counts for 'region':")
display(df['region'].value_counts())

print("\nValue counts for 'country':")
display(df['country'].value_counts())

print("\nValue counts for 'event_type':")
display(df['event_type'].value_counts())

print("\nValue counts for 'disorder_type':")
display(df['disorder_type'].value_counts())


Value counts for 'region':


region
Eastern Africa     129294
Northern Africa     93960
Western Africa      92353
Middle Africa       65031
Southern Africa     33309
Name: count, dtype: int64


Value counts for 'country':


country
Somalia                                         48651
Nigeria                                         40480
Democratic Republic of Congo                    33961
Sudan                                           33287
South Africa                                    22662
Kenya                                           16693
Cameroon                                        16248
Ethiopia                                        14599
South Sudan                                     14422
Morocco                                         13059
Algeria                                         12712
Mali                                            12302
Burkina Faso                                    12079
Libya                                           11898
Egypt                                           11801
Tunisia                                         11203
Burundi                                         10868
Uganda                                           9183
Central African Repu


Value counts for 'event_type':


event_type
Battles                       105556
Violence against civilians    100372
Protests                       94993
Riots                          42737
Strategic developments         38984
Explosions/Remote violence     31305
Name: count, dtype: int64


Value counts for 'disorder_type':


disorder_type
Political violence                    256514
Demonstrations                        115877
Strategic developments                 38984
Political violence; Demonstrations      2572
Name: count, dtype: int64


Missing Values:
event_date                 0
year                       0
disorder_type              0
event_type                 0
sub_event_type             0
actor1                     0
assoc_actor_1         299200
inter1                     0
actor2                111991
assoc_actor_2         328420
inter2                111991
interaction                0
civilian_targeting    290800
iso                        0
region                     0
country                    0
admin1                    17
admin2                  3820
admin3                204055
location                   0
latitude                   0
longitude                  0
geo_precision              0
source                     0
source_scale               0
fatalities                 0
timestamp                  0
dtype: int64


In [25]:
# Summary statistics for numerical columns
print("\nSummary Statistics:")
print(df.describe())  # Mean, min, max, etc., for numerical columns


Summary Statistics:
                year            iso       latitude      longitude  \
count  413947.000000  413947.000000  413947.000000  413947.000000   
mean     2017.527964     510.399988       6.922263      21.886800   
std         6.608510     250.075057      15.495063      16.776814   
min      1997.000000      12.000000     -34.706800     -25.163100   
25%      2015.000000     231.000000       0.315600       8.155500   
50%      2020.000000     566.000000       6.693600      28.043600   
75%      2022.000000     710.000000      13.515700      33.483300   
max      2025.000000     894.000000      37.281500      64.683200   

       geo_precision     fatalities     timestamp  
count  413947.000000  413947.000000  4.139470e+05  
mean        1.279074       2.439153  1.676141e+09  
std         0.494604      24.320917  5.263689e+07  
min         1.000000       0.000000  1.552576e+09  
25%         1.000000       0.000000  1.622068e+09  
50%         1.000000       0.000000  1.689711

#### **Data Cleaning**

### Data Cleaning Plan
> 1. Dropping irrelevant data i.e columns that do not add value to the analysis
> 2. Handling Missing Data Fill or drop missing values based on relevance.
> 3. Removing Duplicates if any exist.
> 4. Detect and remove extreme values that could distort analysis that is remove outliers
> 5. Cleaning Text Data that is renaming columns and correct of typos

NB: for the missing values this is the criteria that will be used:
> * If a column has 0% missing values, no cleaning is needed.
> * If a column has less than 5% missing values, you can either drop the missing rows or fill them using mean/median/mode.
> * If a column has more than 50% missing values, it may be better to drop the column entirely.

#### The following columns will be dropped as they are irrelevant to our anlaysis

> * event_id_cnty – Unique identifier for events (not useful for prediction).
> * timestamp – A Unix timestamp; redundant with event_date.
> * notes – Descriptive text that is difficult to quantify for modeling.
> * source & source_scale – Identifies where the data was collected but may not contribute to predicting future conflict.
> * tags – Sparse data and may not be structured enough for modeling.
> * assoc_actor_1 & assoc_actor_2 – Frequently missing values; may not provide strong predictive power. yet it is backed up by actor
> * iso - The iso column represents the ISO country code, which is a numeric identifier for each country.
> * time_precision - The time_precision column indicates how precise the event date is, hence it less useful.

In [46]:
# Checking for duplicates
# Checking for missing values
print("\n The number of duplicates in the dataset is:")
print(df.duplicated().sum())

# Checking for missing values
print("\nMissing Values:")
print(df.isnull().sum())  # Count missing values in each column


 The number of duplicates in the dataset is:
4291

Missing Values:
event_date                 0
year                       0
disorder_type              0
event_type                 0
sub_event_type             0
actor1                     0
inter1                     0
actor2                111991
inter2                111991
interaction                0
civilian_targeting    290800
region                     0
country                    0
admin1                    17
admin2                  3820
admin3                204055
location                   0
latitude                   0
longitude                  0
geo_precision              0
fatalities                 0
dtype: int64


#### plan to handle missing values

| Column  | Missing values  | Solution and reason                                  |
|---------|-----------------|------------------------------------------------------|
| actor2           | 111,991   | Fill with ***Unknown*** Some events may involve only one actor (e.g., protests, riots)         |
| inter2           | 111,991        | Fill with ***Unknown*** If actor2 is missing, inter2 (interaction type) is also unknown       |
| civilian_targeting | 290,800      | Fill with ***No*** Missing values likely mean civilians were not targeted             |
| admin1           | 17            | ***Drop rows*** only a few missing, so dropping won’t affect the dataset much      |
| admin2           | 3,820         | Fill with ***Unknown*** or mode District info missing, but "Unknown" preserves all events |
| admin3           | 204,055       | ***Drop column*** Too many missing values, making it unreliable for analysis                 |

In [51]:
# fill the rows with missing values
df[['actor2', 'inter2', 'admin2',]] = df[['actor2', 'inter2', 'admin2',]].fillna("Unknown")
df[['civilian_targeting']] = df[['civilian_targeting']].fillna("No")

In [52]:
"""
The civilian_targeting has two entries NO and civilian_targeting, for a better readibility
replacing civilian_targeting with yes will be the enhance the readability of this column
"""
# replace civilian_targeting with Yes
df['civilian_targeting'] = df['civilian_targeting'].replace({'Civilian targeting': 'Yes'})
# preview changes
df['civilian_targeting'].unique()

array(['No', 'Yes'], dtype=object)